packages/boot/test/fast-usdc/fast-usdc.test.ts

```diff
@@ -0,0 +1,110 @@
+  const { walletFactoryDriver: wd } = t.context;
+  const watcherWallet = await wd.provideSmartWallet('agoric1watcher1');
+  t.truthy(watcherWallet);
+});
+
+test.serial(
+  'contract starts; adds to agoricNames; sends invitation',
+  async t => {
+    const {
+      agoricNamesRemotes,
+      evalProposal,
+      buildProposal,
+      refreshAgoricNamesRemotes,
+      storage,
+      walletFactoryDriver: wd,
+    } = t.context;
+
+    const watcherWallet = await wd.provideSmartWallet('agoric1watcher1');
+
+    const materials = buildProposal(
+      '@agoric/builders/scripts/fast-usdc/init-fast-usdc.js',
+      ['--oracle', 'a:agoric1watcher1'],
+    );
+    await evalProposal(materials);
+
+    // update now that fastUSDC is instantiated
+    refreshAgoricNamesRemotes();
+    t.truthy(agoricNamesRemotes.instance.fastUSDC);
+    t.truthy(agoricNamesRemotes.brand.FastLP);
+
+    const { EV } = t.context.runUtils;
+    const agoricNames = await EV.vat('bootstrap').consumeItem('agoricNames');
+    const board = await EV.vat('bootstrap').consumeItem('board');
+    const getBoardAux = async name => {
+      const brand = await EV(agoricNames).lookup('brand', name);
+      const id = await EV(board).getId(brand);
+      t.truthy(storage.data.get(`published.boardAux.${id}`));
+      return unmarshalFromVstorage(
+        storage.data,
+        `published.boardAux.${id}`,
+        makeMarshal().fromCapData,
+        -1,
+      );
+    };
+    t.like(
+      await getBoardAux('FastLP'),
+      {
+        allegedName: 'PoolShares', // misnomer, in some contexts
+        displayInfo: {
+          assetKind: 'nat',
+          decimalPlaces: 6,
+        },
+      },
+      'brand displayInfo available in boardAux',
+    );
+
+    const current = watcherWallet.getCurrentWalletRecord();
+
+    // XXX We should be able to compare objects by identity like this:
+    //
+    // const invitationPurse = current.purses.find(
+    //   p => p.brand === agoricNamesRemotes.brand.Invitation,
+    // );
+    //
+    // But agoricNamesRemotes and walletFactoryDriver
+    // don't share a marshal context.
+    // We should be able to map between them using
+    // const walletStuff = w.fromCapData(a.toCapData(aStuff))
+    // but the marshallers don't even preserve identity within themselves.
+
+    current.purses.length === 1 || Fail`test limited to 1 purse`;
+    const [thePurse] = current.purses;
+    const details = thePurse.balance.value as Array<any>;
+    Array.isArray(details) || Fail`expected SET value`;
+    t.is(details.length, 1, 'oracle wallet has 1 invitation');
+    t.is(details[0].description, 'oracle operator invitation');
+    // XXX t.is(details.instance, agoricNames.instance.fastUSDC) should work
+  },
+);
+
+test.serial('restart contract', async t => {
+  const { EV } = t.context.runUtils;
+  await null;
+  const kit = (await EV.vat('bootstrap').consumeItem(
+    'fastUSDCKit',
+  )) as FastUSDCKit;
```

Comment:
consider updating the type or EconomyBootstrapPowers so this is automatic https://github.com/Agoric/agoric-sdk/blob/8e3d6298bc4275ee25b4ff3b0a896a67a160362d/packages/boot/tools/supports.ts#L53-L58

```suggestion
  const kit = await EV.vat('bootstrap').consumeItem('fastUSDCKit');
```

Reply:
Right.

I tweaked `type ConsumeBootrapItem` instead of `EconomyBootstrapPowers`, though, since the `@agoric/boot` already depends on `@agoric/fast-usdc`.